internal/factsengine/gatherers/sbd.go

```diff
@@ -0,0 +1,94 @@
+package gatherers
+
+import (
+	"os"
+
+	"github.com/hashicorp/go-envparse"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/cluster"
+	"github.com/trento-project/agent/internal/factsengine/entities"
+)
+
+const (
+	SBDConfigGathererName = "sbd_config"
+)
+
+// nolint:gochecknoglobals
+var (
+	SbdConfFileError = entities.FactGatheringError{
+		Type:    "sbd-conf-file-error",
+		Message: "error reading sbd configuration file",
+	}
+
+	SbdConfDecodingError = entities.FactGatheringError{
+		Type:    "sbd-conf-decoding-error",
+		Message: "error decoding configuration file",
+	}
+
+	SbdConfValueNotFoundError = entities.FactGatheringError{
+		Type:    "sbd-conf-value-not-found",
+		Message: "requested field value not found",
+	}
+)
+
+type SBDGatherer struct {
+	configFile string
+}
+
+func NewDefaultSBDGatherer() *SBDGatherer {
+	return NewSBDGatherer(cluster.SBDConfigPath)
+}
+
+func NewSBDGatherer(configFile string) *SBDGatherer {
+	return &SBDGatherer{
+		configFile,
+	}
+}
+
+func (g *SBDGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting SBD Facts gathering")
+
+	conf, err := loadSbdConf(g.configFile)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, requestedFact := range factsRequests {
+		var fact entities.Fact
+		if value, found := conf[requestedFact.Argument]; found {
+			fact = entities.NewFactGatheredWithRequest(requestedFact, &entities.FactValueString{Value: value})
```

Comment:
Most probably we want to use the `entities.ParseStringToFactValue` function here. The values are not always strings. For example, `SBD_WATCHDOG_TIMEOUT` value is an integer